pkg/operator/ceph/csi/controller.go

```diff
@@ -299,11 +305,35 @@ func (r *ReconcileCSI) reconcile(request reconcile.Request) (reconcile.Result, e
 		if err != nil {
 			return opcontroller.ImmediateRetryResult, errors.Wrapf(err, "failed to update CSI driver options for cluster %q", cluster.Name)
 		}
-	}
 
-	err = r.validateAndConfigureDrivers(serverVersion, ownerInfo)
-	if err != nil {
-		return opcontroller.ImmediateRetryResult, errors.Wrap(err, "failed to configure ceph csi")
+		EnableCSIOperator, err = strconv.ParseBool(k8sutil.GetValue(r.opConfig.Parameters, "ROOK_USE_CSI_OPERATOR", "false"))
+		if err != nil {
+			return reconcileResult, errors.Wrap(err, "unable to parse value for 'ROOK_USE_CSI_OPERATOR'")
+		}
+		// Skip the new CSI-operator creation when holder pod is enabled until multus support is added in the CSI operator
+		if EnableCSIOperator && !IsHolderEnabled() {
+			err := r.stopDrivers(serverVersion)
+			if err != nil {
```

Comment:
I focused my review on this section, since it seems the most likely to break existing users. For existing users, I think things look good.

But for multus users, I have a concern. Why we are adding this logic inside the "for each CephCluster" loop instead of after the loop (after all CephClusters are examined)?

I think `IsHolderEnabled()` will be most accurate after the loop processes, which means more accurate determination of whether any cluster is multus. Is there a need to do this inside the loop? Otherwise, I think doing this after the loop would be preferable.

Reply:
We want to create this CR for each cephCluster that's the reason for keeping it inside the loop, I'll double-check again if something can be done and update the code.

Reply:
🤔 I probably shouldn't overthink things here. What you said makes sense. 

Ultimately, for multus users who have holders enabled, we should document that they should not use the experimental CSI driver. If they still do, then we tried our best, and this code will still probably save them from the mistake.